doc/src/manual/types.md

```diff
@@ -1,73 +1,59 @@
-up front are:
-
-  * There is no division between object and non-object values: all values in Julia are true objects
-    having a type that belongs to a single, fully connected type graph, all nodes of which are equally
-    first-class as types.
-  * There is no meaningful concept of a "compile-time type": the only type a value has is its actual
-    type when the program is running. This is called a "run-time type" in object-oriented languages
-    where the combination of static compilation with polymorphism makes this distinction significant.
-  * Only values, not variables, have types -- variables are simply names bound to values, although for
-    simplicity we may say "type of a variable" as shorthand for "type of the value to which a variable refers".
-  * Both abstract and concrete types can be parameterized by other types. They can also be parameterized
-    by symbols, by values of any type for which [`isbits`](@ref) returns true (essentially, things
-    like numbers and bools that are stored like C types or `struct`s with no pointers to other objects),
-    and also by tuples thereof. Type parameters may be omitted when they do not need to be referenced
-    or restricted.
+few drawbacks. Both [abstract types](#man-abstract-types) and [parametric types](#Parametric-Types)
+can be used to create flexible interfaces while leveraging method dispatch and mitigating type errors.
+
+The methods [`isabstracttype`](@ref) and [`isconcretetype`](@ref) can be used
+to check if a type is abstract or concrete.
+They may both return `false` if the type is parametrized but not abstract.
+The [`subtypes`](@ref) and [`supertypes`](@ref) methods
+are also provided for dynamic inspection of the type tree.
+The [`typeof`](@ref) and [`isa`](@ref) methods facilitate checking the type of a value.
 
 Julia's type system is designed to be powerful and expressive, yet clear, intuitive and unobtrusive.
 Many Julia programmers may never feel the need to write code that explicitly uses types. Some
 kinds of programming, however, become clearer, simpler, faster and more robust with declared types.
 
-## Type Declarations
-
-The `::` operator can be used to attach type annotations to expressions and variables in programs.
-There are two primary reasons to do this:
+## Type Annotations
 
-1. As an assertion to help confirm that your program works the way you expect,
-2. To provide extra type information to the compiler, which can then improve performance in some
-   cases
+The default behavior in Julia when type annotations are omitted is to allow values to be of any type.
+When additional expressiveness is needed, however,
+it is easy to gradually introduce explicit annotations into previously "untyped" code.
+Adding annotations serves three primary purposes: to take advantage
+of Julia's powerful multiple-dispatch mechanism, to improve human readability,
+and to catch programmer errors.
```

Comment:
```suggestion
Adding annotations serves four primary purpose: to take advantage
of Julia's powerful multiple-dispatch mechanism, to improve human readability,
to catch programmer errors, and to assist the compiler (especially when defining new types).
```
[Nobody expects the Spanish Inquisition!](https://www.youtube.com/watch?v=Cj8n4MfhjUc)